src/components/buttons/icon-button/icon-button.js

```diff
@@ -33,56 +32,55 @@ const getIconThemeColor = props => {
   return props.icon.props.theme;
 };
 
-const getShapeClassName = shape => styles[shape];
-const getSizeClassName = size => styles[`container-${size}`];
-const getThemeClassName = theme => {
-  if (!theme) return undefined;
-
-  const themeClassName = styles[`theme-${theme}`];
-
-  // Whenever a theme is specified a fitting theme className should
-  // be available except for the default theme.
-  invariant(
-    themeClassName || theme === 'default',
-    `ui-kit/IconButton: the specified theme '${theme}' is not supported.`
-  );
-
-  return themeClassName;
-};
-
 export const IconButton = props => {
-  const dataProps = {
+  const buttonAttributes = {
     'data-track-component': 'IconButton',
     ...filterDataAttributes(props),
   };
+  const isActive = props.isToggleButton && props.isToggled;
   return (
     <div
       onMouseDown={props.handleMouseDown}
       onMouseUp={props.handleMouseUp}
       onMouseOver={props.handleMouseOver}
       onMouseOut={props.handleMouseOut}
-      className={classnames(
-        styles['button-appearance'],
-        getStateClassNames(props.isDisabled, props.isToggled),
-        getShapeClassName(props.shape),
-        getSizeClassName(props.size),
-        getThemeClassName(props.theme)
-      )}
+      css={[
+        css`
+          display: inline-block;
+          background-color: ${vars['--color-white']};
+          box-shadow: ${vars['--shadow-7']};
+          color: ${vars['--color-black']};
+          transition: background-color ${vars['--transition-linear-80ms']},
+            box-shadow 150ms ease-in-out;
+        `,
+        getStateStyles(props.isDisabled, isActive, props.theme),
+        getShapeStyles(props.shape, props.size),
+        getSizeStyles(props.size),
+        getThemeStyles(props.theme),
+      ]}
     >
       <AccessibleButton
-        buttonAttributes={dataProps}
+        buttonAttributes={buttonAttributes}
         type={props.type}
         label={props.label}
         onClick={props.onClick}
+        isToggleButton={props.isToggleButton}
         isToggled={props.isToggled}
         isDisabled={props.isDisabled}
-        className={styles[`container-${props.size}`]}
+        css={getSizeStyles(props.size)}
       >
         <div
-          className={classnames(
-            styles['button-layout'],
-            styles[`container-${props.size}`]
-          )}
+          css={[
+            css`
+              display: flex;
+              align-items: center;
+              justify-content: center;
+              > * + * {
```

Comment:
I find these kinds of styles super weird.

Reply:
Actually there are super useful, that's what the spacings are based for. However, I can check if we need to have to manually apply them in the button components...

Reply:
Sure, but since we don't allow users to pass in children to the buttons, why not just add the spacing to the text and the icon?